File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_joystick'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Joystick Bricklet

ipcon = IPConnection.new # Create IP connection
j = BrickletJoystick.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register pressed callback
j.register_callback(BrickletJoystick::CALLBACK_PRESSED) do
  puts 'Pressed'
end

# Register released callback
j.register_callback(BrickletJoystick::CALLBACK_RELEASED) do
  puts 'Released'
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var j = new Tinkerforge.BrickletJoystick(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

// Register pressed callback
j.on(Tinkerforge.BrickletJoystick.CALLBACK_PRESSED,
    // Callback function for pressed callback
    function () {
        console.log('Pressed');
    }
);

// Register released callback
j.on(Tinkerforge.BrickletJoystick.CALLBACK_RELEASED,
    // Callback function for released callback
    function () {
        console.log('Released');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
